### PR TITLE
Add notice for block patterns new location.

### LINKED ISSUE #41930: Patterns: guide people to new patterns UI
In [Gutenberg v7.7](https://make.wordpress.org/core/2020/03/11/whats-new-in-gutenberg-11-march/) we shipped new block patterns UI (right arrow). In v8.0 we changed its location to block inserter (left arrow).

Screenshot 2020-05-07 at 14 28 54

To avoid customer confusion, educate them for some weeks about the new location by registering a simple header plugin with the same icon that it had before, and opening a popover saying something along the lines _“patterns have moved into the inserter”_.

A simple image of the blue `+` button could be good.

We'd take this out after 1-2 more versions ship. Remember to open an issue to remind us about cleaning it out.

Best place for the code is likely [`wpcom-block-editor`](https://github.com/Automattic/wp-calypso/tree/982d4f15f2e582b8629e97e7bb3b181c3ba78dc1/apps/wpcom-block-editor)

cc @ianstewart

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
#### Changes proposed in this Pull Request

As block patterns were recently moved, there have been a number of support requests from users that have not been able to find them.

This reintroduces the block patterns sidebar plugin to notify users of the change.  This is intended to be temporary and removed in the coming weeks after users have had a chance to adapt to this change.

#### Testing instructions

* Build / Sync FSE plugin on this branch.
* Go to the editor, ensure the block patterns Icon is in the top-right toolbar and in the dropdown menu.
* Verify clicking the icon shows a message that block patterns have moved.
* Verify clicking the example icon opens the block patterns inserter.

<hr/>

Screen Shot 2020-05-07 at 7 36 19 PM
<hr/>

Screen Shot 2020-05-07 at 7 36 37 PM
<hr/>

Screen Shot 2020-05-07 at 8 10 50 PM

<hr/>

Fixes #41930

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/common/index.js [hunk 1]
```diff
@@ -1,4 +1,52 @@
...
+						) }
+					</p>
+					<Button isPrimary icon={ plus } onClick={ openInserter } />
+				</PanelBody>
+			</PluginSidebar>
+		</>
+	);
+};
+
+registerPlugin( 'block-patterns-moved', { render: BlockPatternsMoved } );
```
Comment (simison) [inline apps/full-site-editing/full-site-editing-plugin/common/index.js:52]: Both `plus` and `layout` icons were added in Gutenberg v7.6 (https://github.com/WordPress/gutenberg/pull/20234, https://github.com/WordPress/gutenberg/pull/20181) and patterns were added in v7.7. It's not exact but we could just not register this if those icons are `undefined`? Those users never had patterns to beging with.

Will avoid fatals in older Gutenbergs on Atomic anyway.

File: apps/full-site-editing/full-site-editing-plugin/common/index.js [hunk 2]
```diff
@@ -1,4 +1,52 @@
...
+				{ __( 'Block Patterns Have Moved' ) }
+			</PluginSidebarMoreMenuItem>
+			<PluginSidebar
+				icon={ layout }
+				name={ 'block-patterns-moved' }
+				title={ __( 'Block Patterns Have Moved' ) }
+				className="common__block-patterns-moved"
+			>
+				<PanelBody>
+					<h2>{ __( 'Block Patterns have moved!' ) }</h2>
```
Comment (simison) [inline apps/full-site-editing/full-site-editing-plugin/common/index.js:39]: I think we might need `full-site-editing` textdomain here.

See https://github.com/Automattic/wp-calypso/pull/41966#issuecomment-625709406

**Comment by mtias** [comment]
Nice! I agree a popover should be enough, and not sure we should add the "Have Moved..." to the name / plugin.

File: apps/full-site-editing/full-site-editing-plugin/common/index.js (hunk 1 above)
Comment (sirreal) [reply apps/full-site-editing/full-site-editing-plugin/common/index.js:53]: `icons` is unique, it's bundled rather than external.

**Comment by sirreal** [comment]
There doesn't seem to be a way to render a menu button in the top menu without adding a sidebar. We _can_ render a popover out of the sidebar menu when it's opened.

I've pushed some copy tweaks and added a text domain.

File: apps/full-site-editing/full-site-editing-plugin/common/index.js (hunk 1 above)
Comment (simison) [reply apps/full-site-editing/full-site-editing-plugin/common/index.js:59]: Ahh right, forgot about that. So then that doesn't matter, it's not a biggie to ship this to older versions. Will just increase awareness. :-)

**Comment by Addison-Stavlo** [comment]
> There doesn't seem to be a way to render a menu button in the top menu without adding a sidebar. 

The sidebar plugin seemed the least-hacky way of doing things.  We could create a button with a popover and append it, but that would require setting a timeout interval to wait for the selectors we need to append it to to appear.  Beyond that, we would also need to find a way to append it to the menu as this was another place where users could previously find their block patterns.  Registering a plugin seemed like the simple way to get this set up, but I agree the sidebar is a bit unnecessary.

>We can render a popover out of the sidebar menu when it's opened.

Does a popover add value at that point?

**Comment by sirreal** [comment]
> Does a popover add value at that point?

Probably not, it does point to the component that needs to be clicked, but it's a lot of UI jumping out at once, probably too much.

> There might be some data-dispatch

I looked and the inserter panel opening was local component state. It's not exposed via a data store.

**Comment by simison** [comment]
I'm fine with sidebar — it's not as pretty as popover but does the job.

It's not worth spending too much time on this — let's get it in and get those lost customers use the patterns again.

I agree calling the sidebar just "Patterns" for stylistic reasons.

I'd say we can keep this out for 2 or 3 weeks. Randomly made that up and totes open to other timings. ;-)

**Comment by Addison-Stavlo** [comment]
> I'd say we can keep this out for 2 or 3 weeks. 

That timing makes sense to me.  How long did block patterns live in this sidebar?  a month-ish?  removing it after another ~2 releases seems to make sense.

**Comment by simison** [comment]
> How long did block patterns live in this sidebar?

Since March 25, 2020 (v7.7.1).

Moved to a new spot in May 5, 2020 (v8.0.0).

(via PCYsg-m14-p2)

**Pull request merged**